### PR TITLE
Update to pydantic 2.0

### LINKED ISSUE #472: conda-lock is incompatible with pydantic 2
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

I have conda-lock installed via `brew` (it uses pydantic 2 for the build) and when I enter `conda-lock render --help`, I get the following warnings:

```
❯ conda-lock render --help
/opt/homebrew/Cellar/conda-lock/2.1.1/libexec/lib/python3.11/site-packages/pydantic/_internal/_config.py:261: UserWarning: Valid config keys have changed in V2:
* 'json_encoders' has been removed
  warnings.warn(message, UserWarning)
/opt/homebrew/Cellar/conda-lock/2.1.1/libexec/lib/python3.11/site-packages/pydantic/_internal/_config.py:261: UserWarning: Valid config keys have changed in V2:
* 'allow_mutation' has been removed
  warnings.warn(message, UserWarning)
Usage: conda-lock render [OPTIONS] [LOCK_FILE]
[...]
```

### Conda Info

_No response_

### Conda Config

_No response_

### Conda list

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by mariusvniekerk**
Closes #472

### CODE REVIEW

**Comment by maresb** [comment]
Are you sure we want to do this already?

The Pydantic v2 migration seems like a rough edge. I think this might make it a lot harder for people to install conda-lock in their base environments.

**Comment by maresb** [comment]
Maybe we should instead suppress the warnings for a bit?

**Comment by maresb** [comment]
In light of #462 I think this should be fine.

I'm just mildly concerned because I don't understand the effect of removing

```python
json_encoders = {frozenset: list}
```

**Review by maresb — APPROVED** [review]
I think this deserves its own release

**Comment by maresb** [comment]
@mariusvniekerk, this is marked as draft, but I think it may be ready to merge?

**Comment by mariusvniekerk** [comment]
Yeah just left it as draft so that we can land whatever we want before landing this breaking dependency

**Comment by maresb** [comment]
@mariusvniekerk it looks like I forgot to provide this explanation:

The required new functionality exists in Pydantic 1.10, so pinning `>=2` isn't even necessary. I already added a commit in this branch to relax the pin to `>=1.10`. I tested the changes in #462 by pinning `pydantic <2`.

That should be fine, right?

File: conda_lock/models/__init__.py [hunk 1]
```diff
@@ -1,11 +1,5 @@
 from pydantic import BaseModel
 
 
-class StrictModel(BaseModel):
+class StrictModel(BaseModel, extra="forbid"):
     """A Pydantic BaseModel forbidding extra fields and encoding frozensets as lists"""
```
Comment (mariusvniekerk) [inline conda_lock/models/__init__.py:5]:
```suggestion
    """A Pydantic BaseModel forbidding extra fields"""
```

**Pull request merged**